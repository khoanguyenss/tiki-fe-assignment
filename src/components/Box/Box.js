import React from 'react';
import PropTypes from 'prop-types';
import { BoxStyled } from './Box.styled';

const renderCellData = (data) => {
  if (!data) return '';
  if (data < 0) return 'x';
  return data;
};

const Box = ({ box, openBox }) => {
  if (box.isOpen) {
    return (
      <BoxStyled>
        {renderCellData(box.data)}
      </BoxStyled>
    );
  }
  return (
    <BoxStyled
      isClose
      onClick={
        () => openBox(box)
      }
    />
  );
};

Box.propTypes = {
  box: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
    data: PropTypes.number,
    isOpen: PropTypes.bool,
  }).isRequired,
  openBox: PropTypes.func.isRequired,
};

export default Box;
