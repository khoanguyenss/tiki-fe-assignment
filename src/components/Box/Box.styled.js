import styled from 'styled-components';

export const BoxStyled = styled.div`
  width: 35px;
  height: 35px;
  border: 1px solid #000;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
`;
