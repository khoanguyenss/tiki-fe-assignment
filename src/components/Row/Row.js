import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { RowWrapper } from './Row.styled';
import Box from '../../containers/Box';

const Row = memo(({ row }) => (
  <RowWrapper>
    {row.map((box, index) => <Box box={box} key={index} />)}
  </RowWrapper>
));

Row.propTypes = {
  row: PropTypes.arrayOf(
    PropTypes.shape({
      x: PropTypes.number,
      y: PropTypes.number,
      data: PropTypes.number,
      isOpen: PropTypes.bool,
    }),
  ).isRequired,
};

export default Row;
