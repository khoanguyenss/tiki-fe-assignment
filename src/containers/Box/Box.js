import React from 'react';
import Box from '../../components/Box';
import {
  actions as MineSweeperActions,
} from '../../reducers/minesweeper';
import { useActions } from '../../utils/hooks';

const BoxContainer = (props) => {
  const openBox = useActions(
    MineSweeperActions.openBox,
  );

  return (
    <Box
      {...props}
      openBox={openBox}
    />
  );
};

export default BoxContainer;
