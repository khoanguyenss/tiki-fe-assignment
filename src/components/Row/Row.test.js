import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from '../../store';
import Row from './Row';
import Box from '../Box';

const { store } = configureStore();

describe('<Row />', () => {
  const Component = mount(
    <Provider store={store}>
      <Row
        row={[
          {
            x: 0,
            y: 0,
            data: 0,
            isOpen: false,
          },
          {
            x: 0,
            y: 1,
            data: 0,
            isOpen: false,
          },
          {
            x: 0,
            y: 2,
            data: 0,
            isOpen: false,
          },
          {
            x: 0,
            y: 3,
            data: 0,
            isOpen: false,
          },
        ]}
      />
    </Provider>
    ,
  );

  it('should render', () => {
    expect(Component.length).toBe(1);
  });
  it('match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });
  it('should render 4 boxes', () => {
    const BoxList = Component.find(Box);
    expect(BoxList.length).toBe(4);
  });
});
