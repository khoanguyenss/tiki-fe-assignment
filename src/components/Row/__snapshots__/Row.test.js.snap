// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Memo()
    row={
      Array [
        Object {
          "data": 0,
          "isOpen": false,
          "x": 0,
          "y": 0,
        },
        Object {
          "data": 0,
          "isOpen": false,
          "x": 0,
          "y": 1,
        },
        Object {
          "data": 0,
          "isOpen": false,
          "x": 0,
          "y": 2,
        },
        Object {
          "data": 0,
          "isOpen": false,
          "x": 0,
          "y": 3,
        },
      ]
    }
  >
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "kZAgjJ",
              "rules": Array [
                "
  display: flex;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-bdVaJa kZAgjJ"
        >
          <BoxContainer
            box={
              Object {
                "data": 0,
                "isOpen": false,
                "x": 0,
                "y": 0,
              }
            }
            key="0"
          >
            <Box
              box={
                Object {
                  "data": 0,
                  "isOpen": false,
                  "x": 0,
                  "y": 0,
                }
              }
              gameStatus={0}
              openBox={[Function]}
            >
              <styled.div
                isClose={true}
                isMine={0}
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bwzfXH",
                        "isStatic": false,
                        "lastClassName": "eOxqVQ",
                        "rules": Array [
                          "
  width: 35px;
  height: 35px;
  border: 1px solid #000;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  
  ",
                          [Function],
                          "
  
  ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bwzfXH",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isClose={true}
                  isMine={0}
                  onClick={[Function]}
                >
                  <div
                    className="sc-bwzfXH eOxqVQ"
                    onClick={[Function]}
                  />
                </StyledComponent>
              </styled.div>
            </Box>
          </BoxContainer>
          <BoxContainer
            box={
              Object {
                "data": 0,
                "isOpen": false,
                "x": 0,
                "y": 1,
              }
            }
            key="1"
          >
            <Box
              box={
                Object {
                  "data": 0,
                  "isOpen": false,
                  "x": 0,
                  "y": 1,
                }
              }
              gameStatus={0}
              openBox={[Function]}
            >
              <styled.div
                isClose={true}
                isMine={0}
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bwzfXH",
                        "isStatic": false,
                        "lastClassName": "eOxqVQ",
                        "rules": Array [
                          "
  width: 35px;
  height: 35px;
  border: 1px solid #000;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  
  ",
                          [Function],
                          "
  
  ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bwzfXH",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isClose={true}
                  isMine={0}
                  onClick={[Function]}
                >
                  <div
                    className="sc-bwzfXH eOxqVQ"
                    onClick={[Function]}
                  />
                </StyledComponent>
              </styled.div>
            </Box>
          </BoxContainer>
          <BoxContainer
            box={
              Object {
                "data": 0,
                "isOpen": false,
                "x": 0,
                "y": 2,
              }
            }
            key="2"
          >
            <Box
              box={
                Object {
                  "data": 0,
                  "isOpen": false,
                  "x": 0,
                  "y": 2,
                }
              }
              gameStatus={0}
              openBox={[Function]}
            >
              <styled.div
                isClose={true}
                isMine={0}
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bwzfXH",
                        "isStatic": false,
                        "lastClassName": "eOxqVQ",
                        "rules": Array [
                          "
  width: 35px;
  height: 35px;
  border: 1px solid #000;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  
  ",
                          [Function],
                          "
  
  ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bwzfXH",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isClose={true}
                  isMine={0}
                  onClick={[Function]}
                >
                  <div
                    className="sc-bwzfXH eOxqVQ"
                    onClick={[Function]}
                  />
                </StyledComponent>
              </styled.div>
            </Box>
          </BoxContainer>
          <BoxContainer
            box={
              Object {
                "data": 0,
                "isOpen": false,
                "x": 0,
                "y": 3,
              }
            }
            key="3"
          >
            <Box
              box={
                Object {
                  "data": 0,
                  "isOpen": false,
                  "x": 0,
                  "y": 3,
                }
              }
              gameStatus={0}
              openBox={[Function]}
            >
              <styled.div
                isClose={true}
                isMine={0}
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bwzfXH",
                        "isStatic": false,
                        "lastClassName": "eOxqVQ",
                        "rules": Array [
                          "
  width: 35px;
  height: 35px;
  border: 1px solid #000;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  
  ",
                          [Function],
                          "
  
  ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bwzfXH",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isClose={true}
                  isMine={0}
                  onClick={[Function]}
                >
                  <div
                    className="sc-bwzfXH eOxqVQ"
                    onClick={[Function]}
                  />
                </StyledComponent>
              </styled.div>
            </Box>
          </BoxContainer>
        </div>
      </StyledComponent>
    </styled.div>
  </Memo()>
</Provider>
`;
